core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -84,30 +134,131 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
       }
       if (this.isQuestionCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
-      }
-      if (this.isQuestionCoordinator && this.isTranslationCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_TRANSLATION_SUBMITTER,
-          this.TAB_NAME_TRANSLATION_REVIEWER,
+        this.CONTRIBUTION_TYPES.push(
           this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
+          this.TAB_NAME_QUESTION_REVIEWER);
       }
 
+      this.filter = new ContributorAdminDashboardFilter(
+        [],
+        this.selectedLanguageId,
+        null,
+        this.selectedLastActivity);
+
       this.updateSelectedContributionType(this.CONTRIBUTION_TYPES[0]);
+
+      this.loadingMessage = '';
       this.changeDetectorRef.detectChanges();
     });
 
+    this.lastActivty = [0, 7, 30, 90];
+
     this.contributorDashboardAdminStatsBackendApiService
-      .fetchCommunityStats().then(
+      .fetchTopicsData().then(
         (response) => {
-          this.translationReviewersCount = response.translation_reviewers_count;
-          this.questionReviewersCount = response.question_reviewers_count;
-        });
+          this.topics = response;
+          response.map((
+              topic) =>
+            this.allTopicNames.push(topic.topic));
+          this.applyTopicFilter();
+        }
+      );
+  }
+
+  toggleLanguageDropdown(): void {
+    this.languageDropdownShown = !this.languageDropdownShown;
+  }
+
+  toggleActivityDropdown(): void {
+    this.activityDropdownShown = !this.activityDropdownShown;
+  }
+
+  applyTopicFilter(): void {
+    this.selectedTopicsIds = this.selectedTopics.map((
+        selectedTopic) => {
+      const matchingTopic = this.topics.find(
+        (topicChoice) => topicChoice.topic === selectedTopic);
+      return matchingTopic ? matchingTopic.id : '';
+    });
+    this.filter = new ContributorAdminDashboardFilter(
+      this.selectedTopicsIds,
+      this.selectedLanguageId,
+      null,
+      this.selectedLastActivity);
+  }
+
+  selectLanguage(language: string): void {
+    const currentOption: LangaugeChoice = this.languages.find(
+      (option) => option.language === language) || {
+      language: 'English',
+      id: 'en'
+    };
+    this.selectedLanguage = currentOption.language;
+    this.selectedLanguageId = currentOption.id;
+    this.translationReviewersCount = this.translationReviewersCountByLanguage[
+      this.selectedLanguageId];
+    this.filter = new ContributorAdminDashboardFilter(
+      [],
+      this.selectedLanguageId,
+      null,
+      this.selectedLastActivity);
+    this.languageDropdownShown = false;
+  }
+
+  selectLastActivity(lastActive: number): void {
+    if (lastActive === 0) {
+      this.selectedLastActivity = lastActive;
+      this.filter = new ContributorAdminDashboardFilter(
+        [],
+        this.selectedLanguageId,
+        null);
```

Comment:
0 means there's no filter at all. In frontend it's shown as `All time`. So when the user selects 'All time' last activity filter is removed.

Reply:
Okay well then can we have `ContributorAdminDashboardFilter` correctly handle that logic instead of doing that here? "0" seems like a valid input to the filter, so the filter should handle that correctly.

Reply:
Done. Actually we didn't needed it at all cause the case where value of lastActivity is 0 is already handled in service file.